dvc/output/local.py

```diff
@@ -26,9 +26,10 @@ def _parse_path(self, remote, path):
             # so we should expect both posix and windows style paths.
             # PathInfo accepts both, i.e. / works everywhere, \ only on win.
             #
-            # FIXME: if we have Windows path containig / or posix one with \
+            # FIXME: if we have Windows path containing / or posix one with \
             # then we have #2059 bug and can't really handle that.
             p = self.REMOTE.path_cls(path)
+
```

Comment:
This blank line jumps right into single logical block and thus inhibits readability )

Reply:
matter of taste (and we have different styles everywhere). I'm not opinionated though and it was actually a left-over from the previous fix that was testing if stage is not None (which can be and we test it in other places - let you and @efiop if we need that additional check :) )

Reply:
There is a thin line between a matter of taste and intuition) I am quite sure that both superfluity and lack of empty lines might make people mistake more. 

About missing stages - we might just add an `assert` for now to not permit stageless outs/deps with relative paths.

Reply:
> There is a thin line between a matter of taste and intuition

philosophical question ...  let's not go down this path of discussing how different people perceive information ... may be offline sometime :)

> we might just add an assert 

Tbh, I don't understand the value of asserts at all. In this specific case especially. We are just getting "NPE" instead.

Reply:
The value of asserts is stating clearly what you expect, with an optional message clarifying what and why. Here:
```python
if not p.is_absolute():
    assert self.stage, "Outs with relative paths require stage"
    p = self.stage.wdir / p
```
This way if you ever hit this you'll see a clear message and immediately understand what is going on. Here this is not high value since you get an exception immediately anyway, with a less obvious message, but still. However, on many occasions you check with asserts some invariants, which does not lead to an immediate error but wrong intermediate result, which will later cause all sorts of issues: an incorrect output, an incorrect behavior or a mysterious error in some unrelated code.
```

Reply:
yep, agreed, that in some rare cases they probably can be used to simplify debugging (that's why they were invented in the first place) and as a way to document the code